### PR TITLE
Refactor def and for statements nodes

### PR DESCRIPTION
**Pull request by vladmos**
Def and For nodes are represented in the AST like in the Skylark-in-Go parser (https://github.com/google/skylark/blob/master/syntax/syntax.go).

### CODE REVIEW

**Comment by vladmos** [comment]
Don't think so. These nodes are not enabled by default and noone should rely on the internal structure of the AST. The formatting shouldn't be affected by this change.

**Comment by laurentlb** [comment]
This shouldn't break things. These AST nodes are new and used only in bzl mode (no one should rely on it yet).

File: build/lex.go [hunk 1]
```diff
@@ -844,10 +844,16 @@ func (in *input) assignComments() {
...
+		// Do not assign suffix comments to file or to block statements
 		switch x.(type) {
 		case *File:
 			continue
+		case *DefStmt:
+			continue
+		case *IfStmt:
+			continue
+		case *ForStmt:
+			continue
```
Comment (laurentlb) [inline build/lex.go]: Merge lines: `case *File, *defStmt,` ...

File: build/syntax.go [hunk 2]
```diff
@@ -504,18 +492,18 @@ func (x *ReturnStmt) Span() (start, end Position) {
...
+// A ForStmt represents a for loop block: for x in range(10):.
+type ForStmt struct {
 	Comments
-	Start    Position // position of for
-	LoopVars []Expr
-	Iterable Expr
-	Body     CodeBlock
-	End      // position of the end
+	For  Position // position of for
+	Vars []Expr
```
Comment (laurentlb) [inline build/syntax.go]: Vars should be `Expr`, not `[]Expr`

File: build/syntax.go [hunk 3]
```diff
@@ -472,21 +461,20 @@ func (x *LoadStmt) Span() (start, end Position) {
 	return x.Load, x.Rparen
 }
 
-// A FuncDef represents a function definition expression: def foo(List):.
-type FuncDef struct {
+// A DefStmt represents a function definition expression: def foo(List):.
+type DefStmt struct {
```
Comment (laurentlb) [inline build/syntax.go]: This doesn't match with https://github.com/google/skylark/blob/master/syntax/syntax.go#L140

File: build/syntax.go (hunk 2 above)
Comment (vladmos) [reply build/syntax.go]: Done.

File: build/lex.go (hunk 1 above)
Comment (vladmos) [reply build/lex.go]: Done.

File: build/syntax.go (hunk 3 above)
Comment (vladmos) [reply build/syntax.go]: Done.

**Pull request merged**